cli/pcluster/config/validators.py

```diff
@@ -1253,3 +1253,25 @@ def fsx_ignored_parameters_validator(section_key, section_label, pcluster_config
 def _describe_ec2_key_pair(key_pair_name):
     """Return information about the provided ec2 key pair."""
     return boto3.client("ec2").describe_key_pairs(KeyNames=[key_pair_name])
+
+
+def ebs_volume_iops_validator(section_key, section_label, pcluster_config):
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section("ebs", section_label)
+    volume_size = float(section.get_param_value("volume_size"))
+    volume_type = section.get_param_value("volume_type")
+    volume_type_to_iops_ratio = {"io1": 50, "io2": 500}
+    if section.get_param_value("volume_iops"):
+        volume_iops = float(section.get_param_value("volume_iops"))
+
```

Comment:
I would add a check that, if volume_iops is set in the config and the volume_type does not support specifying a specific IOPS value, then the validator should add an error for that.

this would probably require changing the default value of volume_iops from an int literal to a function that returns 100 if the volume type supports provisioning a specific IOPS value, otherwise it returns None.

Reply:
Haven't done this in this commit. Need another commit to add this check